stdlib/LinearAlgebra/src/symmetriceigen.jl

```diff
@@ -294,3 +294,34 @@ function eigvals!(A::StridedMatrix{T}, F::LU{T,<:StridedMatrix}; sortby::Union{F
     rdiv!(A, U)
     return eigvals!(A; sortby)
 end
+
+
+function eigen(A::Hermitian{Complex{T}, <:Tridiagonal}; kwargs...) where {T}
+    (; dl, d, du) = parent(A)
+    N = length(d)
+    if N <= 1
+        eigen(parent(A); kwargs...)
+    else
+        if A.uplo == 'U'
+            E = du'
+            Er = abs.(du)
+        else
+            E = dl
+            Er = abs.(E)
+        end
+        S = Vector{Complex{T}}(undef, N)
+        S[1] = 1
+        for i ∈ 1:N-1
+            S[i+1] = Er[i] == 0 ? one(T) : S[i] * sign(E[i])
+        end
+        B = SymTridiagonal(real.(d), Er)
```

Comment:
```suggestion
        B = SymTridiagonal(real(d), Er)
```
No dot needed for taking the real part, unless you want to fuse it with other dot calls.